buildSrc/src/main/groovy/org/gradle/build/docs/CacheableAsciidoctorTask.groovy

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.build.docs
+
+import groovy.transform.CompileStatic
+import org.asciidoctor.gradle.AsciidoctorTask
+import org.gradle.api.file.FileTree
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+
+@CompileStatic
+@CacheableTask
+class CacheableAsciidoctorTask extends AsciidoctorTask {
```

Comment:
What happened to `StreamClosingAsciidoctorTask`?

Reply:
This PR is based on release. The `StreamClosingAsciidoctorTask` is on master. I would merge the stream closing part into `CacheableAsciidoctorTask` when I am merging this PR.

It should only be a temporary solution anyway. I am planing to look into that some more to see if I can find the real issue and solve it in a more elegant way. Or produce a upstream fix.